src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory extends StreamFactory
+{
+    /**
+     * Creates a Multipart Stream from parameters and/or files.
+     *
+     * @param array $parameters
+     * @param array $files
+     *
+     * @return StreamInterface
```

Comment:
I do not know if this is off topic or not: When I use the MultipartStreamFactory and the StreamFactory I will (technically) get the same instance back, a `StreamInterface`. In one case I know that my `StreamInterface` is actually a MultipartStream, but what can I do with that object? Can I send it with any of our clients? Will it work without modifying the headers to add a `Content-Type: multipart/form-data; boundary=xxxx`?